src/bp-core/bp-core-rest-api.php

```diff
@@ -392,3 +392,41 @@ function bp_rest_api_v1_dispatch_error( $result, $server, $request ) {
 	);
 }
 add_filter( 'rest_pre_dispatch', 'bp_rest_api_v1_dispatch_error', 10, 3 );
+
+/**
+ * Filter the WP REST API response to return a 403 if the signup feature is disabled.
+ *
+ * @since 15.0.0
+ *
+ * @param mixed           $result Response to replace the requested version with. Can be anything
+ *                                a normal endpoint can return, or null to not hijack the request.
+ * @param WP_REST_Server  $server Server instance.
+ * @param WP_REST_Request $request Request used to generate the response.
+ *
+ * @return mixed
+ */
+function bp_rest_api_signup_disabled_feature_dispatch_error( $result, $server, $request ) {
```

Comment:
Here I'm trying to be helpful to the user. Communicating they need to activate this feature for it to work. The default error message is not very helpful.